Sources/LiveViewNative/LiveView.swift

```diff
@@ -24,114 +22,100 @@ import LVNObjC
+                    case .connectionFailed(let error):
+                        SwiftUI.VStack {
+                            SwiftUI.Text("Connection Failed")
+                                .font(.subheadline)
+                            SwiftUI.Text(error.localizedDescription)
+                        }
+                    }
+                } else {
+                    R.loadingView(for: session.url, state: session.state)
                 }
             }
+        }
+            .task {
+                await session.connect()
+            }
     }
         
     @ViewBuilder
     private var rootNavEntry: some View {
-        if case .enabled = coordinator.config.navigationMode {
-            navigationViewOrStack
-            .introspectNavigationController { navigationController in
-                guard !hasSetupNavigationControllerDelegate else {
-                    return
-                }
-                hasSetupNavigationControllerDelegate = true
-                
-                if let existing = navigationController.delegate {
-                    let proxy = ProxyingNavigationControllerDelegate(first: navigationCoordinator, second: existing)
-                    self.navigationControllerDelegateProxy = proxy
-                    navigationController.delegate = proxy
-                } else {
-                    navigationController.delegate = navigationCoordinator
-                }
-                
-                navigationController.interactivePopGestureRecognizer?.addTarget(navigationCoordinator, action: #selector(NavigationCoordinator.interactivePopRecognized))
-            }
-            .overlay {
-                if navigationCoordinator.state.isAnimating,
-                   !UIAccessibility.prefersCrossFadeTransitions {
-                    GeometryReader { _ in
-                        navHeroOverlayView
-                            .frame(width: navigationCoordinator.currentRect.width, height: navigationCoordinator.currentRect.height)
-                            .clipped()
-                            // if we use the GeometryReader, the offset is with respect to the global origin,
-                            // if not, it's with respect to the center of the screen.
-                            // so, we wrap the view in a GeometryReader, but don't actually use the proxy
-                            .offset(x: navigationCoordinator.currentRect.minX, y: navigationCoordinator.currentRect.minY)
-                            .allowsHitTesting(false)
-                            .animation(navigationCoordinator.state.animation, value: navigationCoordinator.currentRect)
-                    }
-                    .edgesIgnoringSafeArea(.all)
-                }
+        if case .enabled = session.config.navigationMode {
+            if UIDevice.current.userInterfaceIdiom == .pad {
+                navigationSplitView
+            } else {
+                navigationStack
             }
```

Comment:
I don't think using a split view should be automatic. The navigation stack matches the behavior of regular LiveView on the web and the split view is a far enough departure that it should be opt-in.

Reply:
I've made it opt-in via `NavigationMode.splitView`